contracts/base/PeripheryPayments.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.19;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {IERC20Minimal} from "@uniswap/v4-core/contracts/interfaces/external/IERC20Minimal.sol";
+
+import {IPeripheryPayments} from "../interfaces/IPeripheryPayments.sol";
+
+import "../libraries/TransferHelper.sol";
+
+using TransferHelper for address;
+using TransferHelper for IERC20Minimal;
+
+error InsufficientToken();
+
+abstract contract PeripheryPayments is IPeripheryPayments {
+    /// @inheritdoc IPeripheryPayments
+    function sweepToken(address token, uint256 amountMinimum, address recipient) public payable override {
```

Comment:
I think you can merge this and `refundETH` into `sweep(Currency currency, ...)`
and just use CurrencyLibrary nicely here
```
currency.transfer(recipient, currency.balanceOfSelf());
```

Reply:
this abstracts native + ERC20!

Reply:
https://github.com/Uniswap/v4-core/blob/main/src/types/Currency.sol